src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -297,6 +298,111 @@ public AgentSettings LoadSettings()
+            _term.Write(StringUtil.Loc("UninstallingService"));
+            if (_store.IsServiceConfigured())
+            {
+                var serviceControlManager = HostContext.GetService<IServiceControlManager>();
+                try
+                {
+                    serviceControlManager.UnconfigureService();
+                    _term.WriteLine(StringUtil.Loc("Success"));
+                }
+                catch (Exception ex)
+                {
+                    Trace.Error(ex);
+                    _term.WriteLine(StringUtil.Loc("Failed"));
+                    throw;
+                }
+            }
+            else
+            {
+                _term.WriteLine(StringUtil.Loc("NotFound"));
+            }
+
+            //delete agent from the server
+            _term.Write(StringUtil.Loc("UnregisteringAgent"));
+            bool isConfigured = _store.IsConfigured();
+            bool hasCredentials = _store.HasCredentials();
+            if (isConfigured && hasCredentials)
+            {
+                try
+                {
+                    AgentSettings settings = _store.GetSettings();
+                    var credentialManager = HostContext.GetService<ICredentialManager>();
+                    VssCredentials creds = credentialManager.LoadCredentials();
+                    Uri uri = new Uri(settings.ServerUrl);
+                    VssConnection conn = ApiUtil.CreateConnection(uri, creds);
+                    var agentSvr = HostContext.GetService<IAgentServer>();
+                    await agentSvr.ConnectAsync(conn);
+                    await agentSvr.DeleteAgentAsync(settings.PoolId, settings.AgentId);
+                    _term.WriteLine(StringUtil.Loc("Success"));
+                }
+                catch (Exception ex)
+                {
+                    Trace.Error(ex);
+                    bool agentNotFound = ex is TaskAgentPoolNotFoundException || ex is TaskAgentNotFoundException;
+                    if (agentNotFound)
+                    {
+                        _term.WriteLine(StringUtil.Loc("NotFound"));
+                    }
+                    else
+                    {
+                        _term.WriteLine(StringUtil.Loc("Failed"));
+                        throw;
+                    }
+                }
+            }
+            else
+            {
+                _term.WriteLine(StringUtil.Loc("MissingConfig"));
+            }
+
+            //delete credential config file
+            _term.Write(StringUtil.Loc("DeletingCredentials"));
+            if (hasCredentials)
+            {
+                try
+                {
+                    _store.DeleteCredential();
+                    _term.WriteLine(StringUtil.Loc("Success"));
+                }
+                catch (Exception ex)
+                {
+                    Trace.Error(ex);
+                    _term.WriteLine(StringUtil.Loc("Failed"));
```

Comment:
In the begining they see StringUtil.Loc("DeletingCredentials")="Deleting credentials config file: ", and then either "Success.", "Failed." or "Not Found". If it is "Failed", the exception is re-thrown to the parent Agent.ExecuteCommand, which is printing the exception message to the console with _term.WriteError(ex.Message);